### PR TITLE
Fix non-deterministic OpenVINO export failure on NMS models

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Fix non-deterministic OpenVINO export failure on NMS models

**Symptom (surfaced by Ultralytics Platform CPU export telemetry):**

```
OpenVINO: export failure: Couldn't get TorchScript module by tracing.
Tracing failed sanity checks! ERROR: Graphs differed across invocations!
```

intermittently, on NMS-wrapped models (`nms=True`), during OpenVINO export.

### Root cause
`ov.convert_model(model, ...)` is handed a raw `nn.Module`, so OpenVINO's PyTorch frontend runs `torch.jit.trace` internally with its default `check_trace=True`. That re-trace-and-diff sanity check is non-deterministic on the data-dependent control flow of an NMS model (`prim::Constant` ordering varies between trace invocations), so it intermittently raises *"Graphs differed across invocations!"* and the export fails.

### Fix
Pre-trace the model ourselves and hand the resulting `ScriptModule` to `ov.convert_model`, mirroring how the TorchScript and CoreML exporters already trace:

```python
ts = torch.jit.trace(model, im, strict=False, check_trace=False)
ov_model = ov.convert_model(ts, input=None if dynamic else input_shape, example_input=im)
```

`check_trace=False` disables only the non-deterministic re-trace-and-diff sanity check; it does not alter the captured graph. `example_input` and the `input=None if dynamic else input_shape` (dynamic vs static) handling are unchanged, and the INT8 path is upstream-unaffected (`nncf.quantize` consumes the same `ov_model`).

### Verification (openvino 2025.4.1, torch 2.11)
yolo11n OpenVINO export succeeds and the exported model predicts identically to PyTorch (5 detections) across configs:

| config | OpenVINO detections | PyTorch | match |
|---|---|---|---|
| `nms=False` | 5 | 5 | ✓ |
| `nms=True` | 5 | 5 | ✓ |
| `nms=True, dynamic=True` | 5 | 5 | ✓ |
| `nms=True, half=True` | 5 | 5 | ✓ |

No regression across the export surface.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR makes OpenVINO export more reliable by tracing the PyTorch model first and passing that traced model into OpenVINO conversion, avoiding flaky re-tracing failures.

### 📊 Key Changes
- Replaced direct OpenVINO conversion of a raw PyTorch `nn.Module` with conversion from a pre-traced TorchScript model.
- Added `torch.jit.trace(model, im, strict=False, check_trace=False)` before `ov.convert_model(...)`.
- Prevents OpenVINO from internally re-tracing the model with its default graph consistency checks.
- Specifically addresses non-deterministic export failures on models with NMS, which could trigger errors like `"Graphs differed across invocations!"`.
- Included inline comments explaining why the traced model approach is used and how it aligns with existing tracing behavior in other export paths like TorchScript and CoreML.

### 🎯 Purpose & Impact
- ✅ Improves export stability for OpenVINO models, especially for detection pipelines that include NMS.
- ✅ Reduces confusing, intermittent export errors for users exporting Ultralytics models.
- ✅ Makes behavior more consistent across different export formats by reusing a similar tracing approach.
- ✅ Helps developers and users trust OpenVINO export workflows more in real-world deployment scenarios.
- ⚠️ Main impact is on export reliability rather than model features or accuracy; users should see fewer failed conversions rather than changes in inference results.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an engineer will assist shortly 😊 Please review the checklist below to help ensure smooth integration:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The change looks well-scoped and addresses the flaky OpenVINO export path without altering the surrounding export or INT8 flow. I did not find any clear correctness, performance, or maintainability issues in the modified lines based on the provided diff.

**Comment by UltralyticsAssistant** [comment]
Merged — fantastic improvement, @glenn-jocher! 🚀

“Reliability is the precondition for trust.” — Wolfgang Schäuble

This update is a great example of that principle in practice: tracing first and reusing the traced model makes OpenVINO export more dependable, reduces frustrating intermittent failures, and brings better consistency across export paths. Small implementation detail, big real-world impact for users deploying YOLO models.

Thanks for continuing to strengthen the export experience for the whole Ultralytics community.
